Projmgr documentation (#205)
Open-CMSIS-Pack/devtools: tools/projmgr/docs/Manual/Overview.md

    - [Board condition](#board-condition)
    - [Layers in packs](#layers-in-packs)
- [Schema](#schema)

Manual Chapters                          | Content
:----------------------------------------|:-------------------------
[Usage](#usage)                          | Overall Concept, tool setup, and invocation commands
[Project Examples](#project-examples)    | Various example projects to get started
[Project Structure](#project-structure)  | Directory structure of the projects
[YML Input Format](YML-Format.md)        | Format of the various YML input files.

## Revision History

Version            | Description
:------------------|:-------------------------
Draft              | Work in progress

# Overview of Operation

The **csolution - CMSIS Project Manager** processes **User Input Files** (in YML format) and **Software Packs** (in Open-CMSIS-Pack format) to create self-contained CMSIS-Build input files that allow to generate independent projects which may be a part of a more complex application.

The **csolution - CMSIS Project Manager** supports the user with the following features:

- Access to the content of software packs in Open-CMSIS-Pack format to:

>   - Setup the tool chain based on a *Device* or *Board* that is defined in the CMSIS-Packs.

Reviewer: ```suggestion
  - Setup the toolchain based on a *Device* or *Board* that is defined in the CMSIS-Packs.
```

  - Add software components that are provided in the various software packs to the application.
- Organize applications (with a `*.csolution.yml` file) into projects that are independently managed (using `*.cproject.yml` files).
- Manage the resources (memory, peripherals, and user defined) across the entire application to:
  - Partition the resources of the system and create related system and linker configuration.
  - Support in the configuration of software stacks (such as RTOS threads).
  - Hint the user for inclusion of software layers that are pre-configured for typical use cases.
- Organize software layers (with a `*.clayer.yml` file) that enable code reuse across similar applications.
- Manage multiple hardware targets to allow application deployment to different hardware (test board, production hardware, etc.).

> - Manage multiple build types to support software verification (debug build, test build, release build, ect.)

Reviewer: ```suggestion
- Manage multiple build types to support software verification (debug build, test build, release build, etc.)
```

- Before running `csolution` the location of the pack repository shall be set via the environment variable

> `CMSIS_PACK_ROOT` otherwise its default location (todo what is the default?) will be taken.

Reviewer: The default depends on the platform and is given by this function:
https://github.com/Open-CMSIS-Pack/devtools/blob/main/libs/crossplatform/src/CrossPlatformUtils.cpp#L34-L49

> `./<project>/Layer+<target>/<name>` | `*.clayer.yml` and related source files of a layers that are specific to a target have a specific directory.
> `./<project>/Layer/<name>`          | `*.clayer.yml` and related source files of a layers that are common to all targets may have a common directory.

Reviewer: ```suggestion
`./<project>/Layer+<target>/<name>` | `*.clayer.yml` and related source files of a layer that are specific to a target have a specific directory.
`./<project>/Layer/<name>`          | `*.clayer.yml` and related source files of a layer that are common to all targets may have a common directory.
```
